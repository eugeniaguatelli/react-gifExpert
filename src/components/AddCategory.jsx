import { useState } from 'react';
import { PropTypes} from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  // extraigo el argumento event desde onchange 
  const onSubmit= (event) => {
    event.preventDefault();
    if(inputValue.trim().length <1) return;
    onNewCategory(inputValue.trim());
    //limpio desp de agregar
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type="text"
        placeholder="Buscar gif"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
onNewCategory: PropTypes.func.isRequired
}