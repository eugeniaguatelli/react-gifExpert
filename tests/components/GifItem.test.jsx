import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem />', () => { 

    const title= 'Alf' ;
    const url= 'https://alf.com';

    test('should  match snapshot', () => {
        const {container}= render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    
     });

     test('should  show the image with the url and alt', () => {

        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
     });

     test('should  show the title', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
     
     })


 })