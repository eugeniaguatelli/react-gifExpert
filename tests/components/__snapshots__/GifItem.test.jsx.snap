// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should  match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Alf"
      src="https://alf.com"
    />
    <p>
      Alf
    </p>
  </div>
</div>
`;
